### PR TITLE
Test Dynamic Components in a single server context

### PR DESCRIPTION
**Pull request by AndreiTeodorLazar**
#### Description
This PR adds a test for dynamic components in a single server context.

### CODE REVIEW

File: SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/SpatialTestSingleServerDynamicComponents/SpatialTestSingleServerDynamicComponents.cpp [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+#include "Net/UnrealNetwork.h"
+#include "Kismet/GameplayStatics.h"
+
+/**
+ * This test tests dynamic component creation, attachment, removal and replication of properties in a single-server context.
+ *
+ * The test includes 1 Server and 2 Clients.
+ * The flow is as follows:
+ * - Setup:
+ *	- The TestActor is spawned and a dynamic component is immediately created and attached to it.
```
Comment (mironec) [inline SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/SpatialTestSingleServerDynamicComponents/SpatialTestSingleServerDynamicComponents.cpp:16]:
```suggestion
 *  - The TestActor is spawned and a dynamic component is immediately created and attached to it.
```
?

Comment (mironec) [reply SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/SpatialTestSingleServerDynamicComponents/SpatialTestSingleServerDynamicComponents.cpp:16]: Hmm, it's always the first one that gets a tab...

Comment (mironec) [reply SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/SpatialTestSingleServerDynamicComponents/SpatialTestSingleServerDynamicComponents.cpp:16]: Do you have the clang-format thing set up?

Comment (AndreiTeodorLazar) [reply SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/SpatialTestSingleServerDynamicComponents/SpatialTestSingleServerDynamicComponents.cpp:16]: I think I do yeah, I have run the Setup.bat script and AFAIK that should do it, right?

Comment (AndreiTeodorLazar) [reply SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/SpatialTestSingleServerDynamicComponents/SpatialTestSingleServerDynamicComponents.cpp:16]: Is weird that I can't even see that there's a tab, I see it as a 1 space if that makes any sense?

Comment (mironec) [reply SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/SpatialTestSingleServerDynamicComponents/SpatialTestSingleServerDynamicComponents.cpp:16]: that does one part, you can also hook it up to Visual Studio directly, should be somewhere in Slack

Comment (AndreiTeodorLazar) [reply SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/SpatialTestSingleServerDynamicComponents/SpatialTestSingleServerDynamicComponents.cpp:16]: Oh, yes I just found a doc now.

**Review by ImprobableNic — APPROVED** [review]

File: SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/SpatialTestSingleServerDynamicComponents/TestDynamicComponentActor.cpp [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+	DOREPLIFETIME(ATestDynamicComponentActor, OnSpawnComponent);
+	DOREPLIFETIME(ATestDynamicComponentActor, PostInitializeComponent);
+	DOREPLIFETIME(ATestDynamicComponentActor, LateAddedComponent);
+}
+
+void ATestDynamicComponentActor::PostInitializeComponents()
+{
+	Super::PostInitializeComponents();
+
+	if (HasAuthority() && !PostInitializeComponent)
```
Comment (mironec) [inline SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/SpatialTestSingleServerDynamicComponents/TestDynamicComponentActor.cpp:27]:
```suggestion
	if (HasAuthority() && PostInitializeComponent != nullptr)
```
Nit

**Review by mironec — APPROVED** [review]

File: SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/SpatialTestSingleServerDynamicComponents/SpatialTestSingleServerDynamicComponents.cpp [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+		});
+
+	// After a second, the Server sets the references of the PostInitializeComponent and creates and attaches the LateAddedComponent.
+	AddStep(TEXT("SpatialTestSingleServerDynamicComponentsServerAddDynamicComponentsAndReferences"), FWorkerDefinition::Server(1), [this]() -> bool
+		{
+			return GetWorld()->GetTimeSeconds() - TestActor->CreationTime >= 1.0f;
+		}, [this]()
+		{
+			// Make sure the PostInitializeComponent was created and it does not have any reference at this stage.
+			if (!TestActor->PostInitializeComponent || TestActor->PostInitializeComponent->ReferencesArray.Num() != 0)
```
Comment (mironec) [inline SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/SpatialTestSingleServerDynamicComponents/SpatialTestSingleServerDynamicComponents.cpp:63]:
```suggestion
			if (TestActor->PostInitializeComponent == nullptr || TestActor->PostInitializeComponent->ReferencesArray.Num() != 0)
```
Nit, and elsewhere

File: SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/SpatialTestSingleServerDynamicComponents/TestDynamicComponentActor.cpp (hunk 2 above)
Comment (AndreiTeodorLazar) [reply SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/SpatialTestSingleServerDynamicComponents/TestDynamicComponentActor.cpp:27]: Yeah you have mentioned this to me before at some point, forgot about it though, will change it, thank you!

**Pull request merged**